### PR TITLE
refactor(wallet): cleanup and remove unused code in create_tx

### LINKED ISSUE #1710: Cleanup create_tx fee_amount usage
"In create_tx, after fee_amount is used in coin selection it gets assigned new values which end up never being used."

### PR DESCRIPTION
**Pull request by notmandatory**
### Description

Cleanup and remove unused code in `Wallet::create_tx`, this was noticed during review of #1763. See: https://github.com/bitcoindevkit/bdk/pull/1763#discussion_r1876740140

fixes #1710 

### Notes to the reviewers

In addition to removing the unneeded assignments to `fee_amount` and `received` I also refactored creation of the change output to be an `if let` instead of `match` statement since it only needs to do something if there is `Excess::Change`.

I should have done this cleanup as part of #1048.

### Changelog notice

None, only internal cleanup.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Comment by Gudnessuche** [comment]
All tests still pass after doing this?

**Review by oleonardolima — COMMENTED** [review]
Concept ACK a0ba1770c6c34c3afa5148f0b80b2e40753cde02

File: crates/wallet/src/wallet/mod.rs [hunk 1]
```diff
@@ -70,7 +70,7 @@ use crate::types::*;
 use crate::wallet::{
     coin_selection::{
         DefaultCoinSelectionAlgorithm,
-        Excess::{self, Change, NoChange},
+        Excess::{Change, NoChange},
```
Comment (ValuedMammal) [inline crates/wallet/src/wallet/mod.rs]: style nit: Only import the enum and use the explicit `Excess::Change` in the remaining code.

```suggestion
        Excess,
```

**Comment by notmandatory** [comment]
Re-based and fixed `use Excess`.

**Review by oleonardolima — COMMENTED** [review]
cACK cc9050751d655f31c2e51cf9d271747a7b5470de

**Review by ValuedMammal — APPROVED** [review]
ACK cc9050751d655f31c2e51cf9d271747a7b5470de

**Review by ValuedMammal — APPROVED** [review]
Rebased to trigger CI

reACK a2f7a8be

**Pull request merged**